chainstate/src/interface/mod.rs

```diff
@@ -17,5 +17,4 @@ pub mod chainstate_interface;
 pub mod chainstate_interface_impl;
 pub mod chainstate_interface_impl_delegation;
 
-#[cfg(any(test, feature = "mock"))]
 pub mod mock;
```

Comment:
This is an issue that I honestly don't know how to solve. We can't use test-guarded things outside of its crate. Part of me thinks we should do the mock implementations in a separate crate then only import it where its needed as part of `dev-dependencies`. What do you guys think about this?

We can have like a separate crate, call it `mock-impls` (?), where all these impls go. This way it doesn't build with the software every time, except for tests.

Reply:
Nice!

Yeah I was thinking `common` was a good place for them because it's the place the share things, but a separate crate only for `dev-dependencies` is better. Cool, I'll move all of them out as part of this change.

Reply:
Oh, please not common. We try to take things out of it. Only put it in common if there's no other way (if circular dependencies occur)

Reply:
Hehe yeah I figured there was no difference between `common` and exposing it where it was. But I do like the `mock/mock-impls` crate idea

Reply:
So I've only moved `mempool` and `chainstate`, and they should both only be built as dev dependencies . Let me know if there is any issues with this change.

Reply:
Yep. Looking good!